controllers/mattermost/mattermost/mattermost.go

```diff
@@ -103,6 +103,13 @@ func (r *MattermostReconciler) checkMattermostService(
 		return err
 	}
 
+	if current.Spec.Type != desired.Spec.Type && current.Spec.Type != "" {
+		err := r.Resources.DeleteService(types.NamespacedName{Namespace: current.Namespace, Name: current.Name}, reqLogger)
```

Comment:
It would be good to add some log here that we are going to recreate service since the type have changed.

Reply:
Done!